### PR TITLE
Merge editions into activities

### PR DESCRIPTION
**Pull request by mariuscoto**

### CODE REVIEW

**Comment by mariuscoto** [comment]
@Aleks23 @maria please have a look

File: routes/activities.js [hunk 1]
```diff
@@ -11,12 +13,40 @@ exports.index = function(req, res) {
...
+    'name'        : req.body.name,
+    'start'       : new Date(start.substring(7,11), start.substring(3,5), start.substring(0,2)),
+    'end'         : new Date(end.substring(7,11), end.substring(3,5), end.substring(0,2))
+  }
+
+  find = {'_id': mongoose.Types.ObjectId.fromString(req.params.activity)}
+  update = {$push: {'edition': new Edition(new_edition)}}
+  Activity.update(find, update).exec()
+
+  new Edition(new_edition).save()
```
Comment (maria) [inline routes/activities.js]: Again, I would first save the edition and then update the activity. Also, just use one object of type `new Edition()`

File: routes/activities.js [hunk 2]
```diff
@@ -11,12 +13,40 @@ exports.index = function(req, res) {
...
+  start = req.body.start_date
+  end = req.body.end_date
+  var new_edition = {
+    'name'        : req.body.name,
+    'start'       : new Date(start.substring(7,11), start.substring(3,5), start.substring(0,2)),
+    'end'         : new Date(end.substring(7,11), end.substring(3,5), end.substring(0,2))
+  }
+
+  find = {'_id': mongoose.Types.ObjectId.fromString(req.params.activity)}
+  update = {$push: {'edition': new Edition(new_edition)}}
```
Comment (maria) [inline routes/activities.js]: Shouldn't this be an array of editions or something?

File: views/activity.jade [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        br
+        input.datepicker(name='end_date', type='text', required)
+    script.
+      $('.datepicker').datepicker({format: 'dd/mm/yyyy'})
+    br
+    input.btn.btn-success(type='submit', value='Add')
+
+  .spacer
+  h4 List of editions
+  - each ed in activity.edition
```
Comment (maria) [inline views/activity.jade]: Related with the above comments, shouldn't an activity have multiple editions?

**Comment by maria** [comment]
Review done. We should really have some comments in the codebase, it's starting to get bigger and bigger. :+1:

File: routes/activities.js (hunk 2 above)
Comment (fullvlad) [inline routes/activities.js]: It actually is an array of Editions.

File: routes/activities.js (hunk 1 above)
Comment (fullvlad) [inline routes/activities.js]: :+1:

**Pull request merged**